code/modules/mob/living/carbon/human/life.dm

```diff
@@ -91,9 +91,38 @@
 /mob/living/carbon/human/breathe()
 	if(!dna.species.breathe(src))
 		..()
-
+#define HUMAN_MAX_OXYLOSS 3
+#define HUMAN_CRIT_MAX_OXYLOSS (SSmob.wait/30)
 /mob/living/carbon/human/check_breath(datum/gas_mixture/breath)
-	dna.species.check_breath(breath, src)
+
+	var/L = getorganslot("lungs")
+
+	if(!L)
+		if(health >= HEALTH_THRESHOLD_CRIT)
+			adjustOxyLoss(HUMAN_MAX_OXYLOSS + 1)
+		else if(!(NOCRITDAMAGE in dna.species.specflags))
+			adjustOxyLoss(HUMAN_CRIT_MAX_OXYLOSS)
+
+		failed_last_breath = 1
+
+		if(dna && dna.species)
+			var/datum/species/S = dna.species
+
+			if(S.safe_oxygen_min)
+				throw_alert("oxy", /obj/screen/alert/oxy)
+			else if(S.safe_toxins_min)
+				throw_alert("not_enough_tox", /obj/screen/alert/not_enough_tox)
+			else if(S.safe_co2_min)
+				throw_alert("not_enough_co2", /obj/screen/alert/not_enough_co2)
+
+		return 0
+	else
+		if(istype(L,/obj/item/organ/lungs))
```

Comment:
Badmins? Idk.
I just like to do an istype before casting stuff, just in case.
Should I remove this check then?